src/Payum/Core/Bridge/Guzzle/HttpClientFactory.php

```diff
@@ -28,6 +46,6 @@ public static function createGuzzle()
      */
     public static function create()
     {
-        return new HttpClient(static::createGuzzle());
+        return new HttplugClient(HttpClientDiscovery::find());
```

Comment:
this change should be reverted. The whole factory is not used anywhere right now.